subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -168,6 +168,24 @@ int nrf_cloud_uninit(void)
 		}
 	}
 
+#if IS_ENABLED(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD)
+	if (atomic_get(&connection_poll_active)) {
+		uint32_t wait_cnt = 50;
+
+		LOG_DBG("Waiting for connection poll thread to become inactive");
+		while (--wait_cnt && atomic_get(&connection_poll_active)) {
```

Comment:
Would a semaphore with a timeout make more sense than waiting on an atomic like this?

Reply:
i don't really like polling on the value, but it didn't seem necessary to add a new semaphore 🤷